### PR TITLE
PoolOracle ready for audit (?)

### PR DESCRIPTION
**Pull request by bbonanno**

### CODE REVIEW

**Comment by alcueca** [comment]
Minor nitpicks:

PoolOracle.sol L62: Named return not used.
PoolOracle.sol L129: Named return not used.
PoolOracle.sol L144: Typo (ration)

**Comment by alcueca** [comment]
@devtooligan, would you please review this PR?

File: src/oracle/PoolOracle.sol [hunk 1]
```diff
@@ -88,25 +90,27 @@ contract PoolOracle is IPoolOracle {
...
 
     // @inheritdoc IPoolOracle
     function update(address pool) public override {
-        // populate the array with empty observations (oldest call only)
-        // the first time ever that this method is called for a given `pool` we initialise its array of observations
-        for (uint256 i = poolObservations[pool].length; i < granularity; i++) {
-            poolObservations[pool].push();
+        // populate the array with empty observations (only on the first call ever for each pool)
+        unchecked {
+            for (uint256 i = poolObservations[pool].length; i < granularity; ) {
```
Comment (devtooligan) [inline src/oracle/PoolOracle.sol:110]: caching the length here will save a little gas if you're interested, unchecked is the biggest thing though

**Review by devtooligan — APPROVED** [review]
lgtm

File: src/oracle/PoolOracle.sol (hunk 1 above)
Comment (bbonanno) [reply src/oracle/PoolOracle.sol:110]: Didn't bother much as that loop only happens once per pool.

**Pull request merged**